generator/cities_boundaries_builder.cpp

```diff
@@ -102,4 +103,36 @@ bool BuildCitiesBoundaries(string const & dataPath, string const & osmToFeatureP
 
   return true;
 }
+}  // namespace
+
+bool BuildCitiesBoundaries(string const & dataPath, string const & osmToFeaturePath,
+                           OsmIdToBoundariesTable & table)
+{
+  using Mapping = map<uint32_t, vector<osm::Id>>;
+
+  return BuildCitiesBoundaries(dataPath, table, [&]() -> unique_ptr<Mapping> {
+    Mapping mapping;
+    if (!ParseFeatureIdToOsmIdMapping(dataPath + OSM2FEATURE_FILE_EXTENSION, mapping))
+    {
+      LOG(LERROR, ("Can't parse feature id to osm id mapping."));
+      return {};
+    }
+    return make_unique<Mapping>(move(mapping));
+  });
+}
+
+bool BuildCitiesBoundariesForTesting(string const & dataPath, TestIdToBoundariesTable & table)
+{
+  using Mapping = map<uint32_t, vector<uint64_t>>;
+
+  return BuildCitiesBoundaries(dataPath, table, [&]() -> unique_ptr<Mapping> {
+    Mapping mapping;
+    if (!ParseFeatureIdToTestIdMapping(dataPath, mapping))
+    {
+      LOG(LERROR, ("Can't parse feature id to test id mapping."));
+      return {};
```

Comment:
What do you thing is better:
`return {};`
`return nullptr;`
`return make_unique<Mapping>();`
? And why do you prefer `{}`?

Reply:
{} is shorter and more universal, i.e. not depending on the actual return type
make_unique will call a constructor in your example, won't it?

I think we should leave the "{}" as is.

Reply:
Yes. Sure. I just wanted (want) to listen to colleague's opinion on this subject. Thanks.

Reply:
Agree with @mpimenov.
```c++
make_unique
```
doesn't fit here, because I need to return empty smart pointer, not smart pointer to default constructed value.

```
nullptr
```
is boring, because smart pointer has a default constructor.

```
unique_ptr<...>{};
```
is boring too, as it duplicates already specified type of return value, so why do I need to duplicate things?

Therefore I prefer to return just ```{}```. No duplication, exactly what I need and from my point of view is readable, because lambda is short enough.